app/src/main/java/com/example/marcinwisniewski/intivetrainingapp/network/ApiFactory.kt

```diff
@@ -0,0 +1,20 @@
+package com.example.marcinwisniewski.intivetrainingapp.network
+
+import com.example.marcinwisniewski.intivetrainingapp.utils.Constant.Companion.BASE_URL
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.gson.GsonConverterFactory
+import retrofit2.Retrofit
+
+class ApiFactory {
+    companion object {
+        private var retrofitInstance: Retrofit? = null
+        fun getMovieService(): MovieService {
+            if (retrofitInstance == null)
+                retrofitInstance = Retrofit.Builder().baseUrl(BASE_URL)
+                        .addConverterFactory(GsonConverterFactory.create())
+                        .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
+                        .build()
+            return retrofitInstance!!.create(MovieService::class.java)
```

Comment:
avoid !!

Reply:
Fixed( i think so :) )  in first commit after review.